base/broadcast.jl

```diff
@@ -283,6 +288,17 @@ function broadcast_c(f, ::Type{Tuple}, As...)
     n = length(shape[1])
     return ntuple(k->f((_broadcast_getindex(A, k) for A in As)...), n)
 end
+@inline function broadcast_c(f, ::Type{Nullable}, a...)
+    nullity = all(hasvalue, a)
+    S = promote_eltype_op(f, a...)
+    if isleaftype(S) && null_safe_eltype_op(f, a...)
+        Nullable{S}(f(map(unsafe_get, a)...), nullity)
```

Comment:
Are you sure this really gives optimal code? I suspect `map` won't be unrolled. At https://github.com/JuliaStats/NullableArrays.jl/pull/166, I eventually used a generated function for that, and it works well. (That would also allow you to get rid of the `all(::typeof(hasvalue), ...)` hack.)